js/base/Exchange.js

```diff
@@ -2583,7 +2583,7 @@ module.exports = class Exchange {
                 return markets[0];
             }
         }
-        throw new BadSymbol (this.id + ' does not have market symbol ' + symbol);
+        throw new BadSymbol (this.id + ' does not have market symbol ' + this.json (symbol));
```

Comment:
@ttodua this message expects `symbol: string` if that is not the case, it's an entirely different problem that should not be fixed here, we don't want to print out the entire market object or something.

Reply:
but if it expects symbol: string, why it is in `if (typeof symbol...` clause at all? just be raw code, and let `this.market(symbol)`  will fire the exception itself?